test: migrate ERC20 approve redesign test to POM

sleepytanya opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**


Migrate erc20-approve-redesign.spec.ts to POM.



## **Related issues**

Fixes:
https://consensyssoftware.atlassian.net/browse/MMQA-762
## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

sleepytanya committed:
Migrate ERC20 approve redesign test to POM

sleepytanya committed:
Resolve lint issues

sleepytanya committed:
Fix MetaMask string

sleepytanya committed:
Fix lint issues - JSDoc and prettier formatting

sleepytanya committed:
fix: add missing newline at end of file for prettier

sleepytanya committed:
Centralize CSS selectors as constants, make flows use page objects underneath

sleepytanya committed:
Fix linter issues

HowardBraham commented on test/e2e/tests/confirmations/transactions/shared.ts:
```diff
@@ -397,3 +397,23 @@ export async function confirmApproveTransaction(driver: Driver) {
+ * @param driver - The WebDriver instance
+ */
+export async function waitForApproveTransactionWindow(driver: Driver) {
+  await driver.waitUntilXWindowHandles(3);
```
Can we test if this `waitUntilXWindowHandles(3)` is actually necessary?  If it's not needed, it would be better to remove it.

HowardBraham commented on development/circular-deps.jsonc:
The only changes in this file are commas.  I think the problem is you hit `save` and then Prettier formatted it.  Just undo the changes in this file and I'll fix `.prettierignore` elsewhere.

sleepytanya committed:
Revert accidental Prettier formatting of circular-deps.jsonc

sleepytanya committed:
Remove unnecessary waitUntilXWindowHandles(3) call

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Remove redundant verifyTokenInAssetList wrapper function

hjetpoluru commented on test/e2e/page-objects/pages/asset-list.ts:
```diff
@@ -0,0 +1,46 @@
+export class AssetList {
+  // Centralized selectors as constants
+  private static readonly selectors = {
+    tokenListItem: '[data-testid="asset-list-item-',
```
This seems to be a typo. Could you inspect the element and verify?

hjetpoluru commented on test/e2e/page-objects/pages/confirmations/redesign/erc20-approve-transaction-confirmation.ts:
```diff
@@ -0,0 +1,192 @@
+    console.log('Expanding advanced details and verifying all sections');
+
+    // Small delay to ensure UI is ready
+    await this.driver.delay(1000);
```
Could we have specific wait?

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya commented on test/e2e/page-objects/pages/confirmations/redesign/erc20-approve-transaction-confirmation.ts:
```diff
@@ -0,0 +1,192 @@
+    await this.driver.delay(1000);
+
+    // Expand the advanced details section
+    await this.driver.clickElement(
```
Thank you for thorough review, really appreciate that :heart:

sleepytanya committed:
Replace generic delay with specific wait,  updated to use POM selector

sleepytanya committed:
Improved token verification

sleepytanya committed:
Update method name

sleepytanya commented on test/e2e/page-objects/pages/asset-list.ts:
```diff
@@ -0,0 +1,46 @@
+   *
+   * @param tokenSymbol - The symbol of the token to verify (e.g., 'TST', 'HST')
+   */
+  async verifyTokenIsVisible(tokenSymbol: string): Promise<void> {
```
Addressed here: https://github.com/MetaMask/metamask-extension/pull/34108/commits/3400c2ea51233d638b7d8059f65be4804e3d7047

sleepytanya committed:
Remove the similarity to openTokenDetails

sleepytanya commented on test/e2e/page-objects/pages/asset-list.ts:
```diff
@@ -0,0 +1,46 @@
+   *
+   * @param tokenSymbol - The symbol of the token to click
+   */
+  async clickToken(tokenSymbol: string): Promise<void> {
```
Great catch! 
Adressed here: https://github.com/MetaMask/metamask-extension/pull/34108/commits/082aea102ed24ad9201674c4e2ac60717fd1bef1

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Fix race condition in ERC20 approve transaction window

sleepytanya committed:
Remove dead code: unused importAndVerifyToken function

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
fix: restore original simple token import approach for Firefox tests

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
fix: resolve linting issues

sleepytanya committed:
fix: remove tokenSymbol from importTestToken call

- TokenImportConfig type only includes contractAddress and networkChainId
- tokenSymbol no longer needed since verification step was removed

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

hjetpoluru commented on test/e2e/tests/confirmations/transactions/shared.ts:
```diff
@@ -396,3 +396,24 @@ export async function confirmApproveTransaction(driver: Driver) {
+export async function waitForApproveTransactionWindow(driver: Driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+  // Wait for the redesigned confirmation dialog to be fully loaded
+  await driver.waitForSelector('[data-testid="confirm-footer-button"]');
```
This needs to be POM selector.

sleepytanya committed:
Convert createERC20ApproveTransaction to use POM

sleepytanya commented on test/e2e/tests/confirmations/transactions/shared.ts:
```diff
@@ -396,3 +396,24 @@ export async function confirmApproveTransaction(driver: Driver) {
+ */
+export async function createERC20ApproveTransaction(driver: Driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+  await driver.clickElement('#approveTokens');
```
Addressed here: [1863ab3](https://github.com/MetaMask/metamask-extension/pull/34108/commits/1863ab392527f5ceb9a90ab3294ee34d3149e806)

sleepytanya committed:
Convert waitForApproveTransactionWindow to use POM selector

hjetpoluru commented on test/e2e/page-objects/pages/import-tokens-modal.ts:
```diff
@@ -0,0 +1,106 @@
+    importButton: '[data-testid="import-tokens-modal-import-button"]',
+
+    // Network selection
+    networkItemPrefix: '[data-testid="select-network-item-',
```
Looks like there is typo here, could you please check it.

hjetpoluru commented on test/e2e/tests/confirmations/transactions/erc20-approve-redesign.spec.ts:
```diff
@@ -78,100 +81,33 @@ async function mocks(server: MockttpServer) {
 }
 
 async function importTST(driver: Driver) {
```
minor nit - I’m not sure if we still need this step or if we should just call the importTestToken function directly instead.

sleepytanya replied:
[9608c55](https://github.com/MetaMask/metamask-extension/pull/34108/commits/9608c550891822269060f424916b24d44003f303)

hjetpoluru commented on test/e2e/tests/confirmations/transactions/erc20-approve-redesign.spec.ts:
```diff
@@ -78,100 +81,33 @@ async function mocks(server: MockttpServer) {
   });
 }
 
 async function createERC20ApproveTransaction(driver: Driver) {
```
Minor nit - Similar thought on this function too instead we could use this `createERC20ApproveTransactionShared` in the test block directly.

sleepytanya committed:
fix typo

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
fix selector typo properly

sleepytanya committed:
Remove unnecessary delays, removed the dead code function
  importAndVerifyToken, fix unused import issue

sleepytanya commented on test/e2e/page-objects/flows/import-token.flow.ts:
```diff
@@ -0,0 +1,72 @@
+ * @param networkChainId - The chain ID of the network
+ * @param _tokenSymbol - The symbol of the token (unused in original approach)
+ */
+export async function importAndVerifyToken(
```
Thank you! [088ac54](https://github.com/MetaMask/metamask-extension/pull/34108/commits/088ac548388290471faa239258e8161dab837342) - fixed here

sleepytanya committed:
Remove the TokenImportConfig

sleepytanya commented on test/e2e/page-objects/flows/import-token.flow.ts:
```diff
@@ -0,0 +1,72 @@
+ */
+export async function importTestToken(
+  driver: Driver,
+  tokenConfig: TokenImportConfig,
```
addressed here: [5e6de30](https://github.com/MetaMask/metamask-extension/pull/34108/commits/5e6de300bf61c063d9c9aca1d2ca13f28169dd57)

sleepytanya committed:
Relocate openImportTokensModal to AssetListPage

sleepytanya commented on test/e2e/page-objects/pages/home/homepage.ts:
```diff
@@ -43,6 +43,10 @@ class HomePage {
     testId: 'asset-list-control-bar-action-button',
   };
 
+  private readonly importTokensOption = {
```
Addressed in the commits above.

sleepytanya committed:
Remove the
  createERC20ApproveTransaction method

sleepytanya commented on test/e2e/tests/confirmations/transactions/shared.ts:
```diff
@@ -396,3 +398,26 @@ export async function confirmApproveTransaction(driver: Driver) {
+export async function createERC20ApproveTransaction(driver: Driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+  const testDapp = new TestDapp(driver);
+  await testDapp.clickApproveTokens();
```
Addressed here: [b7ee5bc](https://github.com/MetaMask/metamask-extension/pull/34108/commits/b7ee5bcafc6ac82b8f92a8482cee7335469976c7)

sleepytanya committed:
Use assetListPage.openImportTokensModal() to open
  the modal, create an ImportTokensModal instance, call the appropriate import method on the modal

sleepytanya commented on test/e2e/page-objects/pages/import-tokens-modal.ts:
```diff
@@ -0,0 +1,106 @@
+ *
+ * Centralizes all selectors and provides methods to interact with the modal
+ */
+export class ImportTokensModal {
```
[cc21898](https://github.com/MetaMask/metamask-extension/pull/34108/commits/cc2189821559e02e1c7ca4cddbc9c0fcb7921fa5)

sleepytanya committed:
Remove the importTST wrapper
  function

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Remove waitForApproveTransactionWindow and
  replace it with direct calls in the spec

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Address feedback

sleepytanya committed:
Address feedback 1

sleepytanya committed:
Lint

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Feedback

sleepytanya committed:
Fix

sleepytanya committed:
lint

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Merge branch 'main' into pom-migration-erc20-approve

sleepytanya committed:
Fix Import Token timeout

sleepytanya committed:
Lint

sleepytanya committed:
Update with type: anvil

ngalee8311 commented:
> <details><summary>Builds ready [802458c]</summary><ul><li>builds: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-dist-browserify/builds/metamask-chrome-13.0.0.zip">chrome</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-dist-mv2-browserify/builds/metamask-firefox-13.0.0.zip">firefox</a></li>
> <li>builds (beta): <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-beta-browserify/builds/metamask-beta-chrome-13.0.0-beta.0.zip">chrome</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-beta-mv2-browserify/builds/metamask-beta-firefox-13.0.0-beta.0.zip">firefox</a></li>
> <li>builds (flask): <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-flask-browserify/builds/metamask-flask-chrome-13.0.0-flask.0.zip">chrome</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-flask-mv2-browserify/builds/metamask-flask-firefox-13.0.0-flask.0.zip">firefox</a></li>
> <li>builds (test): <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-test-browserify/builds/metamask-chrome-13.0.0.zip">chrome</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-test-mv2-browserify/builds/metamask-firefox-13.0.0.zip">firefox</a></li>
> <li>builds (test-flask): <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-test-flask-browserify/builds/metamask-flask-chrome-13.0.0-flask.0.zip">chrome</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/build-test-flask-mv2-browserify/builds/metamask-flask-firefox-13.0.0-flask.0.zip">firefox</a></li>
> <li>build viz: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/lavamoat-viz/index.html">Build System</a></li>
> <li>bundle size: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/bundle-size/bundle_size.json">Bundle Size Stats</a></li>
> <li>user-actions-benchmark: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/benchmarks/benchmark-chrome-browserify-userActions.json">User Actions Stats</a></li>
> <li>storybook: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/storybook-build/index.html">Storybook</a></li>
> <li>typescript migration: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/ts-migration-dashboard/index.html">Dashboard</a></li>
> <li><a href="https://github.com/MetaMask/metamask-extension/actions/runs/16714966993#artifacts">all artifacts</a></li>
> <li><details>
>        <summary>bundle viz:</summary>
>        <ul><li>background: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-0.html">0</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-1.html">1</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-2.html">2</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-3.html">3</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-4.html">4</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-5.html">5</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-6.html">6</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-7.html">7</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/background-8.html">8</a></li><li>common: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-0.html">0</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-1.html">1</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-2.html">2</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-3.html">3</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-4.html">4</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-5.html">5</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-6.html">6</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-7.html">7</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-8.html">8</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-9.html">9</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-10.html">10</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-11.html">11</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/common-12.html">12</a></li><li>ui: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-0.html">0</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-1.html">1</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-2.html">2</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-3.html">3</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-4.html">4</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-5.html">5</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-6.html">6</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-7.html">7</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-8.html">8</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-9.html">9</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-10.html">10</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-11.html">11</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-12.html">12</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-13.html">13</a>, <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/ui-14.html">14</a></li><li>content-script: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/content-script-0.html">0</a></li><li>offscreen: <a href="https://diuv6g5fj9pvx.cloudfront.net/metamask-extension/16714966993/source-map-explorer/offscreen-0.html">0</a></li></ul>
>      </details></li></ul></details>
> 
> <details><summary>UI Startup Metrics (1130 ± 62 ms)</summary><table><thead><tr><th>Platform</th><th>BuildType</th><th>Page</th><th>Metric</th><th>Mean (ms)</th><th>Min (ms)</th><th>Max (ms)</th><th>Std Dev (ms)</th><th>P 75 (ms)</th><th>P 95 (ms)</th></tr></thead><tbody><tr><td rowspan="22">Chrome</td><td rowspan="11">Browserify</td><td rowspan="11">Home</td><td>uiStartup</td><td align="right">1130</td><td align="right">1032</td><td align="right">1341</td><td align="right">62</td><td align="right">1178</td><td align="right">1227</td></tr><tr><td>load</td><td align="right">956</td><td align="right">857</td><td align="right">1093</td><td align="right">62</td><td align="right">1007</td><td align="right">1067</td></tr><tr><td>domContentLoaded</td><td align="right">947</td><td align="right">851</td><td align="right">1090</td><td align="right">63</td><td align="right">1000</td><td align="right">1061</td></tr><tr><td>domInteractive</td><td align="right">16</td><td align="right">13</td><td align="right">45</td><td align="right">4</td><td align="right">16</td><td align="right">23</td></tr><tr><td>firstPaint</td><td align="right">597</td><td align="right">148</td><td align="right">1080</td><td align="right">372</td><td align="right">967</td><td align="right">1052</td></tr><tr><td>backgroundConnect</td><td align="right">214</td><td align="right">200</td><td align="right">259</td><td align="right">10</td><td align="right">216</td><td align="right">233</td></tr><tr><td>firstReactRender</td><td align="right">21</td><td align="right">13</td><td align="right">172</td><td align="right">16</td><td align="right">21</td><td align="right">29</td></tr><tr><td>getState</td><td align="right">9</td><td align="right">3</td><td align="right">48</td><td align="right">7</td><td align="right">10</td><td align="right">24</td></tr><tr><td>initialActions</td><td align="right">2</td><td align="right">0</td><td align="right">31</td><td align="right">4</td><td align="right">2</td><td align="right">11</td></tr><tr><td>loadScripts</td><td align="right">742</td><td align="right">640</td><td align="right">873</td><td align="right">64</td><td align="right">797</td><td align="right">853</td></tr><tr><td>setupStore</td><td align="right">7</td><td align="right">4</td><td align="right">27</td><td align="right">4</td><td align="right">8</td><td align="right">14</td></tr><tr><td rowspan="11">Webpack</td><td rowspan="11">Home</td><td>uiStartup</td><td align="right">2201</td><td align="right">1651</td><td align="right">2797</td><td align="right">205</td><td align="right">2322</td><td align="right">2535</td></tr><tr><td>load</td><td align="right">1657</td><td align="right">1219</td><td align="right">2109</td><td align="right">159</td><td align="right">1726</td><td align="right">1876</td></tr><tr><td>domContentLoaded</td><td align="right">1650</td><td align="right">1215</td><td align="right">2092</td><td align="right">157</td><td align="right">1722</td><td align="right">1861</td></tr><tr><td>domInteractive</td><td align="right">17</td><td align="right">11</td><td align="right">106</td><td align="right">15</td><td align="right">14</td><td align="right">49</td></tr><tr><td>firstPaint</td><td align="right">201</td><td align="right">61</td><td align="right">1715</td><td align="right">220</td><td align="right">211</td><td align="right">335</td></tr><tr><td>backgroundConnect</td><td align="right">30</td><td align="right">13</td><td align="right">349</td><td align="right">34</td><td align="right">30</td><td align="right">56</td></tr><tr><td>firstReactRender</td><td align="right">211</td><td align="right">90</td><td align="right">394</td><td align="right">117</td><td align="right">372</td><td align="right">388</td></tr><tr><td>getState</td><td align="right">26</td><td align="right">4</td><td align="right">292</td><td align="right">64</td><td align="right">11</td><td align="right">266</td></tr><tr><td>initialActions</td><td align="right">5</td><td align="right">2</td><td align="right">42</td><td align="right">5</td><td align="right">6</td><td align="right">14</td></tr><tr><td>loadScripts</td><td align="right">1646</td><td align="right">1213</td><td align="right">2080</td><td align="right">155</td><td align="right">1718</td><td align="right">1848</td></tr><tr><td>setupStore</td><td align="right">21</td><td align="right">6</td><td align="right">301</td><td align="right">49</td><td align="right">14</td><td align="right">26</td></tr><tr><td rowspan="22">Firefox</td><td rowspan="11">Browserify</td><td rowspan="11">Home</td><td>uiStartup</td><td align="right">1285</td><td align="right">1140</td><td align="right">1598</td><td align="right">89</td><td align="right">1339</td><td align="right">1437</td></tr><tr><td>load</td><td align="right">1113</td><td align="right">982</td><td align="right">1383</td><td align="right">77</td><td align="right">1177</td><td align="right">1249</td></tr><tr><td>domContentLoaded</td><td align="right">1112</td><td align="right">981</td><td align="right">1383</td><td align="right">77</td><td align="right">1177</td><td align="right">1249</td></tr><tr><td>domInteractive</td><td align="right">98</td><td align="right">33</td><td align="right">370</td><td align="right">55</td><td align="right">99</td><td align="right">241</td></tr><tr><td>firstPaint</td><td align="right">NaN</td><td align="right">NaN</td><td align="right">NaN</td><td align="right">NaN</td><td align="right">NaN</td><td align="right">NaN</td></tr><tr><td>backgroundConnect</td><td align="right">25</td><td align="right">15</td><td align="right">72</td><td align="right">10</td><td align="right">28</td><td align="right">49</td></tr><tr><td>firstReactRender</td><td align="right">25</td><td align="right">20</td><td align="right">59</td><td align="right">9</td><td align="right">24</td><td align="right">52</td></tr><tr><td>getState</td><td align="right">6</td><td align="right">2</td><td align="right">41</td><td align="right">7</td><td align="right">5</td><td align="right">25</td></tr><tr><td>initialActions</td><td align="right">3</td><td align="right">0</td><td align="right">47</td><td align="right">6</td><td align="right">2</td><td align="right">13</td></tr><tr><td>loadScripts</td><td align="right">1092</td><td align="right">967</td><td align="right">1367</td><td align="right">77</td><td align="right">1158</td><td align="right">1230</td></tr><tr><td>setupStore</td><td align="right">8</td><td align="right">3</td><td align="right">70</td><td align="right">11</td><td align="right">6</td><td align="right">34</td></tr><tr><td rowspan="11">Webpack</td><td rowspan="11">Home</td><td>uiStartup</td><td align="right">1526</td><td align="right">1326</td><td align="right">1985</td><td align="right">125</td><td align="right">1602</td><td align="right">1775</td></tr><tr><td>load</td><td align="right">1280</td><td align="right">1132</td><td align="right">1682</td><td align="right">110</td><td align="right">1364</td><td align="right">1483</td></tr><tr><td>domContentLoaded</td><td align="right">1280</td><td align="right">1131</td><td align="right">1681</td><td align="right">110</td><td align="right">1363</td><td align="right">1483</td></tr><tr><td>domInteractive</td><td align="right">92</td><td align="right">31</td><td align="right">285</td><td align="right">46</td><td align="right">93</td><td align="right">238</td></tr><tr><td>firstPaint</td><td align="right">NaN</td><td align="right">NaN</td><td align="right">NaN</td><td align="right">NaN</td><td align="right">NaN</td><td align="right">NaN</td></tr><tr><td>backgroundConnect</td><td align="right">31</td><td align="right">17</td><td align="right">119</td><td align="right">18</td><td align="right">31</td><td align="right">73</td></tr><tr><td>firstReactRender</td><td align="right">50</td><td align="right">40</td><td align="right">68</td><td align="right">6</td><td align="right">55</td><td align="right">60</td></tr><tr><td>getState</td><td align="right">10</td><td align="right">2</td><td align="right">96</td><td align="right">14</td><td align="right">6</td><td align="right">52</td></tr><tr><td>initialActions</td><td align="right">3</td><td align="right">0</td><td align="right">23</td><td align="right">5</td><td align="right">3</td><td align="right">15</td></tr><tr><td>loadScripts</td><td align="right">1255</td><td align="right">1113</td><td align="right">1651</td><td align="right">107</td><td align="right">1334</td><td align="right">1460</td></tr><tr><td>setupStore</td><td align="right">9</td><td align="right">4</td><td align="right">44</td><td align="right">7</td><td align="right">7</td><td align="right">25</td></tr></tbody></table>Benchmark value 214 exceeds gate value 10 for chrome browserify home mean backgroundConnect<br>
> Benchmark value 3 exceeds gate value 1 for chrome browserify home mean initialActions<br>
> Benchmark value 234 exceeds gate value 18 for chrome browserify home p95 backgroundConnect<br>
> Benchmark value 11 exceeds gate value 1.2 for chrome browserify home p95 initialActions<br>
> Benchmark value 2202 exceeds gate value 2192 for chrome webpack home mean uiStartup<br>
> Benchmark value 202 exceeds gate value 199 for chrome webpack home mean firstPaint<br>
> Benchmark value 2535 exceeds gate value 2454 for chrome webpack home p95 uiStartup<br>
> Benchmark value 335 exceeds gate value 334 for chrome webpack home p95 firstPaint<br>
> Benchmark value 388 exceeds gate value 370 for chrome webpack home p95 firstReactRender<br>
> Benchmark value 266 exceeds gate value 195 for chrome webpack home p95 getState<br>
> Benchmark value 14 exceeds gate value 7 for chrome webpack home p95 initialActions<br>
> Benchmark value 26 exceeds gate value 25 for firefox browserify home mean backgroundConnect<br>
> Benchmark value 26 exceeds gate value 25 for firefox browserify home mean firstReactRender<br>
> Benchmark value 3 exceeds gate value 1 for firefox browserify home mean initialActions<br>
> Benchmark value 241 exceeds gate value 195 for firefox browserify home p95 domInteractive<br>
> Benchmark value 25 exceeds gate value 24 for firefox browserify home p95 getState<br>
> Benchmark value 13 exceeds gate value 2 for firefox browserify home p95 initialActions<br>
> Benchmark value 34 exceeds gate value 27 for firefox browserify home p95 setupStore<br>
> Benchmark value 31 exceeds gate value 26 for firefox webpack home mean backgroundConnect<br>
> Benchmark value 51 exceeds gate value 38 for firefox webpack home mean firstReactRender<br>
> Benchmark value 4 exceeds gate value 1 for firefox webpack home mean initialActions<br>
> Benchmark value 238 exceeds gate value 156 for firefox webpack home p95 domInteractive<br>
> Benchmark value 73 exceeds gate value 49 for firefox webpack home p95 backgroundConnect<br>
> Benchmark value 60 exceeds gate value 50 for firefox webpack home p95 firstReactRender<br>
> Benchmark value 52 exceeds gate value 32 for firefox webpack home p95 getState<br>
> Benchmark value 15 exceeds gate value 2 for firefox webpack home p95 initialActions<br>
> <b>Sum of mean exceeds: 244ms | Sum of p95 exceeds: 617.8ms<br>
> Sum of all benchmark exceeds: 861.8ms</b><br>
> </details>
> 
> <details><summary>Bundle size diffs [🚨 Warning! Bundle size has increased!]</summary><ul><li>background: 43 Bytes (0%)</li>
> <li>ui: 0 Bytes (0%)</li>
> <li>common: 10 Bytes (0%)</li></ul></details>
> 
>
